static/js/stream_create.js

```diff
@@ -129,10 +137,16 @@ function update_announce_stream_state() {
         privacy_type === "invite-only" || privacy_type === "invite-only-public-history";
     $announce_stream_label.removeClass("control-label-disabled");
 
+    if (!$announce_stream_checkbox.prop("disabled")) {
+        stream_announce_previous_value = $announce_stream_checkbox.prop("checked");
+    }
```

Comment:
I suspect this logic will produce unexpected results when clicking between different types of private streams or between different types of public streams. We ideally want to only store this value when changing from public to private, and restore it when doing the opposite.

I'm not entirely sure I can come up with a problematic reproducer easily, but FYI.

Reply:
For avoiding saving values when changing between different type of private streams only, I have added the `disabled` check since we only want to store the previous value when the checkbox is enabled and it is enabled only when the privacy is public or web-public. So, we only store the value if the previous privacy value was public or web-public.

Also, when clicking between public and web-public, I don't think there will be problem. Suppose the initial privacy selected is web-public and then the user unchecks the announce checkbox, then on changing the privacy to public we just again try to uncheck the checkbox (which ideally we should not since it is already in that state, but cannot figure out a clean way to prevent those).

Reply:
It took me a while to understand this logic as correct (and I ended up adding some print statements before understanding it), but I now agree it is. I added some comments that I think should explain it clearly for future readers (or at least for future me!).